node/src/components/chain_synchronizer.rs

```diff
@@ -57,15 +48,9 @@ pub(crate) enum JoiningOutcome {
     ShouldExitForUpgrade,
     /// We finished initial synchronizing, with the given block header being the result of the fast
     /// sync task.
-    Synced { highest_block_header: BlockHeader },
-    /// We ran `commit_genesis` or `commit_upgrade` and created the given switch block immediately
-    /// afterwards. `highest_block_header` will be the same as that in
-    /// `block_and_execution_effects` except where we synced using a trusted block of the last
-    /// switch block before an emergency upgrade, in which case it might be a later block.
-    RanUpgradeOrGenesis {
-        block_and_execution_effects: BlockAndExecutionEffects,
-        validators_to_sign_immediate_switch_block: HashSet<PublicKey>,
-        highest_block_header: BlockHeader,
+    Synced {
+        highest_block_header: Box<BlockHeader>,
+        maybe_immediate_switch_block_data: Option<Box<ImmediateSwitchBlockData>>,
```

Comment:
Can you add a comment about this field? It's missing from the comment on the enum variant.

Reply:
Are we even using this in `chain_synchronizer`? Or are we just passing it on to be used in the `Participating` reactor? Then we might not need the detour via this module?

Reply:
I think it's just passed to the `Participating` reactor. I think you have a point here - since IIRC the `Joiner` reactor retains the `ChainspecLoader` instance, so this could be read directly from that instance when transitioning to `Participating`.